coral-spark/src/main/java/com/linkedin/coral/spark/CoralSpark.java

```diff
@@ -70,6 +70,22 @@ public static CoralSpark create(RelNode irRelNode) {
     return new CoralSpark(baseTables, sparkUDFInfos, sparkSQL);
   }
 
+  /**
+   * Users use this function to get CoralSqlNode from CoralRelNode
+   * This should be used when user need to get CoralSqlNode from CoralRelNode by applying
+   * spark specific transformations on CoralRelNode
+   * with Coral-schema output schema
+   *
+   * @return [[SqlNode]]
+   */
+  public static SqlNode getCoralSqlNode(RelNode irRelNode) {
```

Comment:
I'm unclear why this is needed. 

CoralSpark API should take Coral intermediate representation as input and return Spark SQL / SparkSQLNode. 

You could create a 
```
 String convert(SqlCommand sqlCommand, RelNode coralRelNode) {
// generate CoralSqlNode (RHS) for coralRelNode
// set the updated CoralSqlNode (RHS) in sqlCommand
// send this (combined) SqlNode to CoralSqlNodeToSparkSqlNodeConverter and then SparkSqlRewriter
}
```